### PR TITLE
Flashing process improvements

### PR DESCRIPTION
**Pull request by algernon**
This is a fairly big change, aimed at making the flashing process not only smoother, but more reliable too. The biggest factor in this is that after resetting the keyboard, while waiting for it to come back up in bootloader mode, we now try to re-detect which serial port it is on, instead of relying on it appearing as the same port. This alone makes the process a whole lot more reliable.

The PR also adds support for using an external (but still bundled) program for flashing, and makes this configurable on the Preferences screen (defaulting to off, though).

Fixes - among other things - #365.

This is currently a draft, to make it easier to review with pre-built binaries. If/when approved, I'll drop the topmost commit and mark the PR ready.

### CODE REVIEW

**Comment by algernon** [comment]
Also, this is on top of #507.

File: .travis.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ matrix:
     - os: windows
 env:
   - YARN_GPG=no
-if: branch = master OR tag IS present
+if: branch = master OR tag IS present OR branch = windows/avrdude
```
Comment (obra) [inline .travis.yml:18]: back this one out?

**Comment by obra** [comment]
+1 to merge once you back out the one commit and fix the travis config.

**Comment by algernon** [comment]
That last commit is the travis one. ;)

 Backed it out, merging.

**Pull request merged**